components/CommonComponents/ShellBox/index.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import styles from './index.module.scss';
+
+interface Props {
+  textToCopy: string;
+}
+
+const ShellBox = ({ children, textToCopy }: React.PropsWithChildren<Props>) => {
+  const [copied, copyText] = useCopyToClipboard();
+
+  const handleCopyCode = async (event: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
Shouldn't the children be the text to copy? Afaik you can simple convert the `children.toString()` no?

Reply:
it's use like that.
```tsx
<ShellBox textToCopy="choco install nvs">
      <span className={styles.installTextNoSelect}>$</span>
      <span className={styles.installTextCommand}>choco </span>install nvs
    </ShellBox>
```

So we don't touch to that .

Reply:
Just because it was done like that doesn't mean we cannot change it.

I believe the textToCopy could be an optional argument and by default it goes to the approach I mentioned before? Wdyt

Reply:
What's the difference of CodeBox and ShellBox then? Just the style?

Reply:
No, codebox it's use to display code an hightlight it. ShellBox it's just an box for an shell command.

Reply:
You do realise semantically these are the same. Both display code, both allow code to be copied. Both have syntax highlighting to a sort.

Hence virtually both are the same. The only real difference are the layouts. Hence in UX that's what we call "flavours"

I do understand you have a different view of whatever you think you know, but it's kinda irritating from you to reject/disagree with these points when you have no real understanding of these concepts.

Instead of simply saying no, what about researching about these concepts and reflect a little about it. Because honestly, for me it makes sense that both are the same component but with different styles.

The same applies to the textToCopy issue. You simply negated my comment without a proper reasoning of why we should keep the prop. I'd recommend for you to actually start reflecting on your decisions and if I make a suggestions it must be because there's a point on it. I wouldn't recommend something just out of the blue. I don't want to sound rude/arrogant, but it's kinda tiring to have these kind of discussions because they're not productive.

Reply:
I didn't say to use `children.toString()` (It was just an example), you should use `ReactDomServer.renderToString`.

> To conclude we have to keep textToCopy in props

What kind of conclusion is that?

Reply:
> I didn't say to use children.toString() (It was just an example), you should probably use ReactDomServer.renderToString.

This case can work but. we will have this code copied:
`$ curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash`

Reply:
The idea of having `textToCopy` is to have the copied text different from the displayed text.

Reply:
I know what the component is about. Again, I think you're facing some creativity issues here and not understanding the broader goal of what I'm talking about having sub components / flavours.

I mentioned before that we he textToCopy could be an optional parameter and would default to the children.

Also, the only issue there is that it would copy the "$" symbol. This is easily solvable if we remove the "$" symbol from the original children and simply use CSS's :before or :after selectors to decorate each line of the ShellBox with the $ symbol.

The TL;DR is that there are many ways of achieving what I'm referring to, and you're simply creating a storm on a glass of water, by saying this is impossible.